### PR TITLE
Implement createTable functionality

### LINKED ISSUE #55: Implement createTable on Catalog
# Description

Implement the method for `createTable` and `createExternalTable` on the `Catalog`. 

- [createTable](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.Catalog.createTable.html#pyspark.sql.Catalog.createTable)
- [createExternalTable](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.Catalog.createExternalTable.html#pyspark.sql.Catalog.createExternalTable)

### PR DESCRIPTION
**Pull request by lexara-prime-ai**
# Description
This pull request introduces the `create_table` and `create_external_table` methods to the `catalog` module:
- Added `create_table` method to catalog for creating managed tables.
- Added `create_external_table` method to catalog for creating external tables.
- Added tests for managed and external table creation with schema validation and data assertions.

# Related Issue(s)
- closes #55 

# Documentation
* [createTable](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.Catalog.createTable.html#pyspark.sql.Catalog.createTable)
* [createExternalTable](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.Catalog.createExternalTable.html#pyspark.sql.Catalog.createExternalTable)

### CODE REVIEW

**Comment by sjrusso8** [comment]
@lexara-prime-ai  going forward can you make separate branches or squash your old changes? right now it says there are 36 commits for this change, but it's actually only < 5

**Comment by sjrusso8** [comment]
@lexara-prime-ai  can you run `cargo clippy` over your changes? It's part of the precommit config, and clippy is showing some errors and recommendations when I run it over your branch

**Review by sjrusso8 — CHANGES_REQUESTED** [review]
Hey here are changes that you should look at

File: core/src/catalog.rs [hunk 1]
```diff
@@ -141,6 +144,95 @@ impl Catalog {
...
+    pub async fn create_table(
+        self,
+        table_name: &str,
+        source: Option<&str>,
+        schema: Option<DataType>,
+        description: Option<&str>,
+        options: Option<HashMap<String, String>>,
+    ) -> Result<DataFrame, SparkError> {
+        let source = if let Some(s) = source {
+            s.to_string()
```
[the remark this answers is not in the document]
Comment (lexara-prime-ai) [reply core/src/catalog.rs:156]: @sjrusso8 that makes sense, I noticed the default source was set to parquet when running manual tests but thanks for confirming. Now we can confidently remove that step.

**Comment by lexara-prime-ai** [comment]
> @lexara-prime-ai can you run `cargo clippy` over your changes? It's part of the precommit config, and clippy is showing some errors and recommendations when I run it over your branch

Doing that right away. I also pulled in some changes from upstream as well. We could've avoided commit pollution. I'll see what I can do.

File: core/src/catalog.rs [hunk 2]
```diff
@@ -141,6 +144,95 @@ impl Catalog {
...
+            description,
+            options: options.unwrap_or_default(),
+            path: None,
+        };
+
+        let cat_type = Some(spark::catalog::CatType::CreateTable(create_table_message));
+
+        let rel_type = spark::relation::RelType::Catalog(spark::Catalog { cat_type });
+
+        let plan = LogicalPlanBuilder::from(rel_type).plan_root();
```
[the remark this answers is not in the document]
Comment (lexara-prime-ai) [reply core/src/catalog.rs:187]: This is definitely a lot cleaner. Thanks for the pointer.

**Comment by lexara-prime-ai** [comment]
> @lexara-prime-ai going forward can you make separate branches or squash your old changes? right now it says there are 36 commits for this change, but it's actually only < 5

I had a separate branch but pulled all updates into `main`. Maybe we can re-open the pr from said branch after resolving the change requests.

**Pull request closed without merging**